test/StakeManager.js

```diff
@@ -1204,5 +1205,43 @@ describe('StakeManager', function () {
       const tx = stakeManager.connect(signers[3]).withdraw(epoch, stakerIdacc3);
       await assertRevert(tx, 'Nonpositive Stake');
     });
+
+    it('staker should not be able to withdraw more than his sToken balance', async function () {
+      let epoch = await getEpoch();
+      await razor.connect(signers[12]).approve(stakeManager.address, tokenAmount('420000'));
+      await stakeManager.connect(signers[12]).stake(epoch, tokenAmount('420000'));
+      const stakerId = await stakeManager.stakerIds(signers[12].address);
+      const staker = await stakeManager.stakers(stakerId);
+      const sToken = await stakedToken.attach(staker.tokenAddress);
+      await stakeManager.connect(signers[12]).unstake(epoch, stakerId, tokenAmount('420000'));
+      await sToken.connect(signers[12]).transfer(signers[10].address, tokenAmount('20000'));
+      for (let i = 0; i < WITHDRAW_LOCK_PERIOD; i++) {
+        await mineToNextEpoch();
+      }
+      epoch = await getEpoch();
+      const tx = stakeManager.connect(signers[12]).withdraw(epoch, stakerId);
+      await assertRevert(tx, 'locked amount lost');
+      await stakeManager.connect(signers[12]).resetLock(stakerId);
+    });
+
+    it('sToken burn should fail if staker calls resetLock and his sToken balance is less than sAmount', async function () {
```

Comment:
Please rename this to 
'ResetLock should fail, if delegators balance is less than the amount to be penalized'

Reply:
Here also "ResetLock should fail, if stakers sRZR balance is less than the amount to be penalized"

Reply:
cool